extensions-core/histogram/src/main/java/io/druid/query/aggregation/histogram/ApproximateHistogram.java

```diff
@@ -1581,7 +1581,7 @@ public double sum(final float b)
           z = (-b + Math.sqrt(b * b - 4 * a * c)) / (2 * a);
         }
         final double uj = this.positions[i - 1] + (this.positions[i] - this.positions[i - 1]) * z;
-        quantiles[j] = (float) uj;
+        quantiles[j] = ((float) uj < this.max()) ? (float) uj : this.max();
```

Comment:
It's nice how simple, yet effective this tweak is. Would it make sense to do a similar tweak for mins?

Reply:
In addition, if you have a clearly worded explanation of why the uj's can be greater than the max, a comment would be useful. Something about why it's necessary to clamp them to the max.